File: Mediator/template_cpp/ConcreteMediator.cpp
#include "ConcreteMediator.h"

// Send a message, cant send message to itself
void ConcreteMediator::sendMessage(Colleague *sender, const std::string &message) {
    for (Colleague* colleague : colleagues) {
        if (colleague != sender) {
            colleague->receiveMessage(message);
        }
    }
}

File: Mediator/template_java/ConcreteMediator.java
import java.util.ArrayList;
import java.util.List;

// Definition of the concrete mediator that has a list of collegues its responsible for and can send messages
class ConcreteMediator implements Mediator {
    private List<Colleague> colleagues = new ArrayList<>();

    public void addColleague(Colleague colleague) {
        colleagues.add(colleague);
    }

    @Override
    public void sendMessage(Colleague sender, String message) {
        for (Colleague colleague : colleagues) {
            if (colleague != sender) {
                colleague.receiveMessage(message);
            }
        }
    }
}
